File: EngTaskGradingNetBESolution/EngTaskGradingNetBE/Models/Dtos/TaskDto.cs
namespace EngTaskGradingNetBE.Models.Dtos
{
  public record TaskDto(
          int Id,
          string Title,
          string? Keywords,
          string? Description,
          int? MinGrade,
          string Aggregation
      );

  public record TaskCreateDto(
          string Title,
          int CourseId,
          string? Keywords,
          string? Description,
          int? MinGrade,
          string Aggregation
      );

  public record TaskUpdateDto
  (
    string Title,
    string? Keywords,
    string? Description,
    int? MinGrade,
    string Aggregation
  );
}

File: eng-task-grading-react-fe/src/model/task-dto.ts
export interface TaskDto {
  id: number;
  title: string;
  description?: string;
  keywords: string | null;
  minGrade: number | null;
  aggregation: "min" | "max" | "avg" | "last";
}

export interface TaskCreateDto {
  courseId: number;
  title: string;
  description: string | null;
  keywords: string | null;
  minGrade: number | null;
  aggregation: "min" | "max" | "avg" | "last";
}

export interface TaskUpdateDto {
  title: string;
  description: string | null;
  keywords: string | null;
  minGrade: number | null;
  aggregation: "min" | "max" | "avg" | "last";
}